cmd/editorconfig-checker/main.go

```diff
@@ -155,7 +161,22 @@ func parseArguments() {
 func main() {
 	parseArguments()
 
+	if cpuprofile != "" {
+		f, err := os.Create(cpuprofile)
+		if err != nil {
+			currentConfig.Logger.Error("Creating CPU profile file %s: %v", cpuprofile, err.Error())
+			exitProxy(exitCodeErrorOccurred)
+		}
+		if err := pprof.StartCPUProfile(f); err != nil {
+			currentConfig.Logger.Error("Starting CPU profile: %v", err.Error())
+			exitProxy(exitCodeErrorOccurred)
+		}
+	}
+
 	config := *currentConfig
+	// force the exclude regexp to be compiled and cached
+	config.CachedExcludesAsRegexp()
```

Comment:
If I'm not wrong, this is the the first call to this method.

So it would cache.

But the regexp could fail, and right now it would panic

So here I would use this


```suggestion
	_, err = config.CachedExcludesAsRegexp()
	if err != nil {
		currentConfig.Logger.Error("Caching exclusions: %v", err.Error())
		exitProxy(exitCodeErrorOccurred)
	}
```

It comes with this

https://github.com/editorconfig-checker/editorconfig-checker/pull/488#discussion_r2311912998